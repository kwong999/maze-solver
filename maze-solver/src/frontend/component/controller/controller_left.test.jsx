import React from 'react';
import renderer from 'react-test-renderer';
import ControllerLeft from './controller_left';
jest.mock('./build_board', () => 'BuildBoard');
jest.mock('./tile_option', () => 'TileOption');


describe('controller_left', () => {
  const component = renderer.create(
    <ControllerLeft
      dimensionRow={'dimensionRow'}
      dimensionCol={'dimensionCol'}
      handleChange={'handleChange'}
      handleBuildBoard={'handleBuildBoard'}
      changeTileType={'changeTileType'}
      tileType={'tileType'}
    />
  )
  test('test should render without error', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test('child component BuildBoard has props', () => {
    let componentInstance = component.root;
    let buildBoardProps = componentInstance.findByType('BuildBoard').props;
    expect(buildBoardProps.dimensionRow).toMatch('dimensionRow');
    expect(buildBoardProps.dimensionCol).toMatch('dimensionCol');
    expect(buildBoardProps.handleChange).toMatch('handleChange');
    expect(buildBoardProps.handleBuildBoard).toMatch('handleBuildBoard');
  })

  test('child component TileOption has props', () => {
    let componentInstance = component.root;
    let tileOptionProps = componentInstance.findByType('TileOption').props;
    expect(tileOptionProps.tileType).toMatch('tileType');
    expect(tileOptionProps.changeTileType).toMatch('changeTileType');
  })
})