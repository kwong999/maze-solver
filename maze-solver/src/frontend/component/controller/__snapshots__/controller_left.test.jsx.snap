// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controller_left test should render without error 1`] = `
<div
  className="controller-left"
>
  <BuildBoard
    dimensionCol="dimensionCol"
    dimensionRow="dimensionRow"
    handleBuildBoard="handleBuildBoard"
    handleChange="handleChange"
  />
  <TileOption
    changeTileType="changeTileType"
    tileType="tileType"
  />
</div>
`;
